lib/browser/chrome-extension.js

```diff
@@ -1,533 +1,540 @@
 'use strict'
 
-const { app, webContents, BrowserWindow } = require('electron')
-const { getAllWebContents } = process.electronBinding('web_contents')
-const ipcMainUtils = require('@electron/internal/browser/ipc-main-internal-utils')
-
-const { Buffer } = require('buffer')
-const fs = require('fs')
-const path = require('path')
-const url = require('url')
-const util = require('util')
-
-// Mapping between extensionId(hostname) and manifest.
-const manifestMap = {} // extensionId => manifest
-const manifestNameMap = {} // name => manifest
-const devToolsExtensionNames = new Set()
-
-const generateExtensionIdFromName = function (name) {
-  return name.replace(/[\W_]+/g, '-').toLowerCase()
-}
+if (process.electronBinding('features').isExtensionsEnabled()) {
```

Comment:
Can't we just not execute this file at all if extension support is enabled instead of wrapping the whole thing in an `if`?

Reply:
Hm... I was avoiding refactoring the things that call this but I suppose that's doable.